types/es-abstract/helpers/isPropertyDescriptor.d.ts

```diff
@@ -1,14 +1,12 @@
-import {
-    AccessorDescriptor as ESAccessorDescriptor,
-    DataDescriptor as ESDataDescriptor,
-    PropertyDescriptor as ESPropertyDescriptor,
-} from '../index';
+import { PropertyDescriptor as ESPropertyDescriptor } from '../index';
 
 declare function IsPropertyDescriptor(
     ES: {
+        // This is necessary to avoid "TS2345: Object literal may only specify known properties":
+        [method: string]: any;
         Type(O: unknown): string | undefined;
-        IsAccessorDescriptor(Desc: unknown): Desc is ESAccessorDescriptor;
-        IsDataDescriptor(Desc: unknown): Desc is ESDataDescriptor;
+        IsAccessorDescriptor(Desc: unknown): boolean;
```

Comment:
why remove the predicate syntax?

Reply:
Because it fixes some assignability issues when using these types in untyped **JavaScript** code.

Reply:
untyped JS code is not impacted by any TS information, so i don't understand why that would make a difference.

Reply:
@ExE-Boss can you help me understand what issues are caused by using the predicate syntax here?

Reply:
It happens when importing the v1.17 module and trying to assign `es‑abstract/*/IsAccessorDescriptor` to it, because its type defaults to `(Desc: any) => boolean`.

Reply:
sooo should the typeof `IsAccessorDescriptor` then be also fixed to be `(Desc: unknown) => Desc is ESAccessorDescriptor`?

Reply:
`require("es‑abstract").IsAccessorDescriptor` has the correct `(Desc: unknown) => Desc is ESAccessorDescriptor` type.

It’s the **v1.17**’s `require("es‑abstract/{5,201*}/IsAccessorDescriptor.js")` that’s currently untyped.

Reply:
ok - so can we add types for that so that this type can be better?

Reply:
`require("es‑abstract/{5,201*}/IsAccessorDescriptor.js")` will be handled in <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44805>.